java/client/src/main/java/glide/utils/ArrayTransformUtils.java

```diff
@@ -75,6 +75,28 @@ public static <T, U extends T> U[] castArray(T[] objectArr, Class<U> clazz) {
                 .toArray(size -> (U[]) Array.newInstance(clazz, size));
     }
 
+    /**
+     * Casts an <code>Object[][]</code> to <code>T[][]</code> by casting each nested array and every
+     * array element.
+     *
+     * @param outerObjectArr Array of arrays of objects to cast.
+     * @param clazz The class of the array elements to cast to.
+     * @return An array of arrays of type U, containing the elements from the input array.
+     * @param <T> The base type from which the elements are being cast.
+     * @param <U> The subtype of T to which the elements are cast.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T, U extends T> U[][] castArrayofArrays(
+            T[] outerObjectArr, Class<U> clazz, Class<U[]> arrClazz) {
+        if (outerObjectArr == null) {
+            return null;
+        }
+        for (int i = 0; i < outerObjectArr.length; i++) {
+            outerObjectArr[i] = (T) castArray((T[]) outerObjectArr[i], clazz);
+        }
+        return castArray(outerObjectArr, arrClazz);
```

Comment:
I'd optimize it a bit
```suggestion
    public static <T, U extends T> U[][] castArrayofArrays(
            T[] outerObjectArr, Class<U> clazz) {
        if (outerObjectArr == null) {
            return null;
        }
        for (int i = 0; i < outerObjectArr.length; i++) {
            outerObjectArr[i] = (T) castArray((U[]) outerObjectArr[i], clazz);
        }
        return castArray(outerObjectArr, (Class<U[]>) clazz.arrayType());
```

Update usage and javadoc too

Reply:
arrayType() seems to be a Java 12+ feature, since I was using openjdk11 it did not exist under Class, should we consider that and keep it less optimized or is a post java 12 version recommended over 11?

Reply:
use hivemind https://stackoverflow.com/a/61509635/21176342

Reply:
I was on that post and scrolled right past that after looking at the first answer, oops